llvm/lib/Target/AMDGPU/SMInstructions.td

```diff
@@ -29,6 +29,7 @@ class SM_Pseudo <string opName, dag outs, dag ins, string asmOps, list<dag> patt
   let mayStore = 0;
   let mayLoad = 1;
   let hasSideEffects = 0;
+  let maybeAtomic = 0;
```

Comment:
This preserves the existing behaviour but it feels wrong. In practice it means that SIMemoryLegalizer does not set glc on volatile SMEM loads.

Reply:
Probably should fix this in a separate change

Reply:
#77443